dm/syncer/err-operator/operator.go

```diff
@@ -209,22 +192,26 @@ func (h *Holder) MatchAndApply(startLocation, endLocation binlog.Location, realE
 		// set LogPos as start position
 		for _, ev := range operator.events {
 			ev.Header.LogPos = startLocation.Position.Pos
-			ev.Header.Timestamp = realEventHeaderTS
+			ev.Header.Timestamp = currentEvent.Header.Timestamp
 			if e, ok := ev.Event.(*replication.QueryEvent); ok {
 				if startLocation.GetGTID() != nil {
 					e.GSet = startLocation.GetGTID().Origin()
 				}
 			}
 		}
 
-		// set the last replace event as end position
-		e := operator.events[len(operator.events)-1]
-		e.Header.EventSize = endLocation.Position.Pos - startLocation.Position.Pos
-		e.Header.LogPos = endLocation.Position.Pos
-		if e, ok := e.Event.(*replication.QueryEvent); ok {
-			if endLocation.GetGTID() != nil {
-				e.GSet = endLocation.GetGTID().Origin()
+		if operator.op == pb.ErrorOp_Replace {
+			// set the last replace event as end position
+			e := operator.events[len(operator.events)-1]
+			e.Header.EventSize = endLocation.Position.Pos - startLocation.Position.Pos
+			e.Header.LogPos = endLocation.Position.Pos
+			if e, ok := e.Event.(*replication.QueryEvent); ok {
+				if endLocation.GetGTID() != nil {
+					e.GSet = endLocation.GetGTID().Origin()
+				}
 			}
+		} else {
```

Comment:
better use clear check (elseif pb.ErrorOp_Inject) because we have many other operations

Reply:
ok